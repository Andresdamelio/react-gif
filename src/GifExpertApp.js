import React, { useState } from 'react';
import './index.css';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';
import { v4 as uuidv4 } from 'uuid';
const GifExpertApp = () => {
  const [categories, setCategories] = useState(['Rick and morty']);

  return (
    <>
      <h1 className='title animate__animated animate__fadeIn'>
        Gif Expert App
      </h1>
      <AddCategory setCategories={setCategories} />
      <hr />

      {categories.map((category) => {
        return <GifGrid key={uuidv4()} category={category} />;
      })}
    </>
  );
};

export default GifExpertApp;
