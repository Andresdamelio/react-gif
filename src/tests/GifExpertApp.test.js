import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Probando el funcionamiento de <GifExpertApp />', () => {
  test('Probando que <GifExpertApp /> se renderiza correctamente', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('<GifExpertApp /> debe de mostrar una lista de categorias correctamente', () => {
    const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];

    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
