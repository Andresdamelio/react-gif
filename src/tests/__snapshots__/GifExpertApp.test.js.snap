// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el funcionamiento de <GifExpertApp /> <GifExpertApp /> debe de mostrar una lista de categorias correctamente 1`] = `
<Fragment>
  <h1
    className="title animate__animated animate__fadeIn"
  >
    Gif Expert App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="One Punch"
    key="b734ba2e-9214-4d25-b0b9-79862a15fab9"
  />
  <GifGrid
    category="Samurai X"
    key="a6376d6e-5652-4a43-a126-4d708cf3d446"
  />
  <GifGrid
    category="Dragon Ball"
    key="01f32e85-5a07-484d-9a6b-671340606802"
  />
</Fragment>
`;

exports[`Probando el funcionamiento de <GifExpertApp /> Probando que <GifExpertApp /> se renderiza correctamente 1`] = `
<Fragment>
  <h1
    className="title animate__animated animate__fadeIn"
  >
    Gif Expert App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;
